### PR TITLE
Add makefile for running tests, License file, badges, etc.

### PR DESCRIPTION
**Pull request by pritesh**

### CODE REVIEW

**Comment by flashvoid** [comment]
Looks fine.

File: README.md [hunk 1]
```diff
@@ -99,4 +101,15 @@ the cluster and how to restart the services:
     sudo service romana-agent start
     ```    
 
-
+### Previous Releases
+Previous Releases can be found [here][github-release]
```
Comment (jbrendel) [inline README.md]: Forgot the period at the end of the sentence.

File: agent/agent.go [hunk 2]
```diff
@@ -65,10 +65,10 @@ func (a *Agent) SetConfig(config common.ServiceConfig) error {
 func (a *Agent) Routes() common.Routes {
 	routes := common.Routes{
 		common.Route{
-			"POST",
-			"/",
-			a.index,
-			func() interface{} {
+			Method:  "POST",
```
Comment (jbrendel) [inline agent/agent.go]: Nice! The names are helpful.

File: agent/agent.go [hunk 3]
```diff
@@ -96,7 +96,7 @@ func Run(rootServiceURL string) (chan common.ServiceMessage, string, error) {
 
 }
 
-// Implements Name() method of Service interface.
+// Name Implements method of Service interface.
```
Comment (jbrendel) [inline agent/agent.go]: Lower-case 'implements'.

File: agent/netif.go [hunk 4]
```diff
@@ -28,9 +28,10 @@ import (
 type NetIf struct {
 	Name string `form:"interface_name" mapstructure:"interface_name"`
 	Mac  string `form:"mac_address" mapstructure:"interface_name"`
-	IP  net.IP `form:"ip_address" mapstructure:"ip_address"`
+	IP   net.IP `form:"ip_address" mapstructure:"ip_address"`
 }
 
+// SetIP Parse and set IP.
```
Comment (jbrendel) [inline agent/netif.go]: "SetIP parses and sets the IP address of the interface."

File: ipam/ipam.go [hunk 5]
```diff
@@ -40,18 +40,18 @@ const (
 func (ipam *IPAMSvc) Routes() common.Routes {
 	routes := common.Routes{
 		common.Route{
-			"POST",
-			"/vms",
-			ipam.addVm,
-			func() interface{} {
+			Method:  "POST",
```
Comment (jbrendel) [inline ipam/ipam.go]: Can we have the same, nice alignment of the values as below?

File: root/root.go [hunk 6]
```diff
@@ -115,27 +115,26 @@ func (root *Root) handleConfig(input interface{}, ctx common.RestContext) (inter
...
+			Pattern:     "/config/{serviceName}",
+			Handler:     root.handleConfig,
+			MakeMessage: nil,
 		},
 		common.Route{
-			"POST",
-			"/config/{serviceName}/port",
-			root.handlePortUpdate,
-			func() interface{} {
+			Method:  "POST",
```
Comment (jbrendel) [inline root/root.go]: Can we have the same alignment of the values as in the other structures?

File: tenant/tenant.go [hunk 7]
```diff
@@ -39,44 +39,44 @@ const (
 func (tsvc *TenantSvc) Routes() common.Routes {
 	routes := common.Routes{
 		common.Route{
-			"POST",
-			tenantsPath,
-			tsvc.addTenant,
-			func() interface{} {
+			Method:  "POST",
```
Comment (jbrendel) [inline tenant/tenant.go]: Value alignment?

File: tenant/tenant.go [hunk 8]
```diff
@@ -39,44 +39,44 @@ const (
...
+			Pattern:     tenantsPath,
+			Handler:     tsvc.listTenants,
+			MakeMessage: nil,
 		},
 		common.Route{
-			"POST",
-			tenantsPath + "/{tenantId}" + segmentsPath,
-			tsvc.addSegment,
-			func() interface{} {
+			Method:  "POST",
```
Comment (jbrendel) [inline tenant/tenant.go]: Value alignment?

File: topology/topology.go [hunk 9]
```diff
@@ -47,36 +47,36 @@ const (
...
+			Pattern:     hostListPath,
+			Handler:     topology.handleHostListGet,
+			MakeMessage: nil,
 		},
 		common.Route{
-			"POST",
-			hostListPath,
-			topology.handleHostListPost,
-			func() interface{} {
+			Method:  "POST",
```
Comment (jbrendel) [inline topology/topology.go]: Value alignment?

**Comment by jbrendel** [comment]
Otherwise ok.

**Pull request merged**